moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_display.cpp

```diff
@@ -305,6 +305,13 @@ void MotionPlanningDisplay::reset()
   query_robot_goal_->setVisible(query_goal_state_property_->getBool());
 }
 
+void MotionPlanningDisplay::setName(const QString& name)
+{
+  BoolProperty::setName(name);
```

Comment:
The problem is, that the MotionPlanning Panel is not a "associated widget", so the functions in the rviz::Display class will not change the name of the Panel but only the name of the Display.
There was an issue why the MotionPlanning Panel can't be a "associated widget". See here #442 and here #452.